src/scripts/storage/clipperData.ts

```diff
@@ -0,0 +1,49 @@
+import {GetResponseAsync, TimeStampedData} from "../http/cachedHttp";
+import {ClipperCachedHttp} from "../http/clipperCachedHttp";
+
+import {Logger} from "../logging/logger";
+
+import {ClipperStorageGateStrategy} from "./clipperStorageGateStrategy";
+import {Storage} from "./storage";
+import {StorageGateStrategy} from "./StorageGateStrategy";
+
+/**
+ * The data-access-like object that handles the fetching and caching of data
+ * from Clipper-specific endpoints (TODO: a factory class that is responsible for
+ * providing getRemoteValue objects?). Also provides a gating mechanism as a
+ * sanity-check for what information gets stored locally.
+ */
+export class ClipperData implements Storage {
+	private storage: Storage;
+	private storageGateStrategy: StorageGateStrategy;
+	private cachedHttp: ClipperCachedHttp;
+
+	constructor(storage: Storage, logger?: Logger) {
+		this.storage = storage;
+		this.storageGateStrategy = new ClipperStorageGateStrategy(storage);
+		// We pass 'this' as the Storage object as it handles all the sanity-check gating in the setValue
+		this.cachedHttp = new ClipperCachedHttp(this, logger);
+	}
+
+	public setLogger(logger: Logger) {
+		this.cachedHttp.setLogger(logger);
+	}
+
+	public getAndCacheFreshValue(key: string, getRemoteValue: GetResponseAsync, updateInterval?: number): Promise<TimeStampedData> {
```

Comment:
Would love to hear what other people think. I had struggles trying to find a good name for this method, so I would love to hear everyone's opinion :)
Renaming it back to getFreshValue is definitely an option if we feel it's better than this one